### PR TITLE
fix(protocol): revert impl deployment V2

### PR DESCRIPTION
**Pull request by adaki2004**
Almost same as https://github.com/taikoxyz/taiko-mono/pull/14611/ but this is now a combined version so that we can make it TransparentProxyUpgradeable.

Please @cyberhorsey  or @davidtaikocha confirm if this is also a working solution (`invokeMessageCall() `succeeds when bridging erc20 from L1 to L2) and then we can go forward with this type of solution and abandon the other.

### CODE REVIEW

File: packages/protocol/contracts/tokenvault/ERC20Vault.sol [hunk 1]
```diff
@@ -330,13 +330,21 @@ contract ERC20Vault is
         private
         returns (address btoken)
     {
-        ProxiedBridgedERC20 bridgedToken = new ProxiedBridgedERC20();
+        address bridgedToken = Create2Upgradeable.deploy({
+            amount: 0, // amount of Ether to send
+            salt: keccak256(
```
Comment (dantaik) [inline packages/protocol/contracts/tokenvault/ERC20Vault.sol:335]: bytes32 salt = ...

Comment (dantaik) [reply packages/protocol/contracts/tokenvault/ERC20Vault.sol:335]: Yes, and maybe just simply:
```bytes32 salt = keccak256(abi.encode(ctoken))```

**Comment by cyberhorsey** [comment]
Will attempt today and confirm if this one also works.

**Comment by cyberhorsey** [comment]
@adaki2004 it worked.
https://explorer.internal.taiko.xyz/tx/0x2dd4f9987320677730d75c23b620463087f5f8f7ff4911d2777469cc2ca6122b
here is a TX of it beign deployed, with a fresh ERC20, with the latest changes from this branch as the ERC20Vault on L2.

**Review by dantaik — DISMISSED** [review]

File: packages/protocol/contracts/tokenvault/ERC20Vault.sol [hunk 2]
```diff
@@ -268,6 +268,16 @@ contract ERC20Vault is
         return interfaceId == type(IRecallableMessageSender).interfaceId;
     }
 
+    /// @dev Encodes sending bridged or canonical ERC20 tokens to the user.
+    /// @param user The user's address.
+    /// @param token The token address.
+    /// @param to To address.
+    /// @param amount Amount to be sent.
+    /// @return msgData Encoded message data.
+    /// @return _amount Token amount left in the contract. (Todo Daniel: why we
```
[the remark this answers is not in the document]
Comment (dantaik) [reply packages/protocol/contracts/tokenvault/ERC20Vault.sol]: See this: https://github.com/taikoxyz/taiko-mono/pull/14621/commits/05aef61b06b0cb189ca2fbd56b3765695818773f

**Review by dantaik — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**